app/cdap/components/GridTable/index.tsx

```diff
@@ -206,9 +199,17 @@ export default function GridTable({ handleTransformationUpload, storeData }) {
         setShowBreadCrumb={setShowBreadCrumb}
         showBreadCrumb={showBreadCrumb}
         columnType={dataprep.types[selectedColumn]}
-        submitMenuOption={(option, datatype) =>
-          !transformationOptions.includes(option) ? onMenuOptionSelection(option, datatype) : null
-        }
+        hasSimilarType={checkAllColumnsHasSimilarType(Object.values(dataprep.types))}
+        allColumnsType={Object.values(dataprep.types)[0] as string}
+        submitMenuOption={(option, datatype) => {
```

Comment:
move the callback to a separate function

Reply:
moved to seperate function